### PR TITLE
feat(mssql-cdc): add ssl support for sqlserver-cdc

### PR DESCRIPTION
**Pull request by KeXiangWang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As titled.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

At https://github.com/risingwavelabs/risingwave-docs/blob/38a20f52702649437c5d5eb62dd41551da230f69/docs/guides/ingest-from-sqlserver-cdc.md?plain=1#L118,
please add one more line
```
| database.encrypt | Optional. Specify whether you want to enable SSL encryption. Currently, `trustServerCertificate` is enabled, regardless of the value of `database.encrypt`. |
```

### CODE REVIEW

File: src/connector/src/source/cdc/external/mod.rs [hunk 1]
```diff
@@ -243,6 +243,12 @@ pub struct ExternalTableConfig {
     #[serde(rename = "ssl.root.cert")]
     #[serde(alias = "debezium.database.sslrootcert")]
     pub ssl_root_cert: Option<String>,
+
+    /// `encrypt` specifies whether connect to SQL Server using SSL.
+    /// Choices include `true`, `false`.
+    /// This field is optional.
+    #[serde(rename = "database.encrypt", default = "Default::default")]
+    pub encrypt: String,
```
Comment (StrikeW) [inline src/connector/src/source/cdc/external/mod.rs:251]: optional field should wrap in `Optional`.

File: java/connector-node/risingwave-connector-service/src/main/resources/sql_server.properties [hunk 2]
```diff
@@ -21,4 +21,5 @@ name=${hostname}:${port}:${database.name}.${schema.name}.${table.name:-RW_CDC_Sh
 # In sharing cdc mode, transaction metadata will be enabled in frontend.
 # For sql server, it's always false actually.
 provide.transaction.metadata=${transactional:-false}
-database.encrypt=false
+database.encrypt=${database.encrypt:-false}
+database.trustServerCertificate=true
```
Comment (StrikeW) [inline java/connector-node/risingwave-connector-service/src/main/resources/sql_server.properties:25]: It is correct to always to `true` for this config? maybe we can default to `true` and allow passing different value.

**Review by StrikeW — COMMENTED** [review]
Rest lgtm

File: java/connector-node/risingwave-connector-service/src/main/resources/sql_server.properties (hunk 2 above)
Comment (KeXiangWang) [reply java/connector-node/risingwave-connector-service/src/main/resources/sql_server.properties:25]: Ideally, we should allow users to either enable `trustServerCertificate` or provide a path to the cert. We can support it later.

File: src/connector/src/source/cdc/external/mod.rs (hunk 1 above)
Comment (KeXiangWang) [reply src/connector/src/source/cdc/external/mod.rs:250]: Updated.

**Review by StrikeW — APPROVED** [review]

**Pull request merged**